gemcalibration/include/gem/calib/Calibration.h

```diff
@@ -0,0 +1,224 @@
+                {"phaseMax"  , "Phase max (int)"  },
+                {"stepSize", "Step size (int)"},
+                {"armDacPoins", "ARM DAC points"}, 
+            };
+
+            std::map<std::string, uint32_t> m_amcOpticalLinks;
+
+            dacScanType_t m_dacScanType;
+
+            struct dacScanTypeFeature{
+                std::string label;
+                std::map<std::string, uint32_t> range;
+            };
+            
+             /**
+             *  map of selectable DAC scan for the VFAT3 parameters and relative labels and range limits
+             */
+
+            
+            std::map<dacScanType_t, dacScanTypeFeature> m_dacScanTypeParams{
+                {CFG_CAL_DAC,{"CFG_CAL_DAC", {{"CFG_CAL_DAC_Min"  , 0},{"CFG_CAL_DAC_Max", 255},}}},
+                {CFG_BIAS_PRE_I_BIT, {"CFG_BIAS_PRE_I_BIT",{{"CFG_BIAS_PRE_I_BIT_Min",0},{"CFG_BIAS_PRE_I_BIT_Max", 255},}}}, 
+                {CFG_BIAS_PRE_I_BLCC,{"CFG_BIAS_PRE_I_BLCC",{{"CFG_BIAS_PRE_I_BLCC_Min", 0},{"CFG_BIAS_PRE_I_BLCC_Max", 63},}}},
+                {CFG_BIAS_PRE_I_BSF,{"CFG_BIAS_PRE_I_BSF",{{"CFG_BIAS_PRE_I_BSF_Min",0}, {"CFG_BIAS_PRE_I_BSF_Max", 63},}}},
+                {CFG_BIAS_SH_I_BFCAS,{"CFG_BIAS_SH_I_BFCAS",{{"CFG_BIAS_SH_I_BFCAS_Min",0},{"CFG_BIAS_SH_I_BFCAS_Max", 255},}}},
+                {CFG_BIAS_SH_I_BDIFF,{"CFG_BIAS_SH_I_BDIFF",{{"CFG_BIAS_SH_I_BDIFF_Min", 0},{"CFG_BIAS_SH_I_BDIFF_Max", 255},}}},
+                {CFG_BIAS_SD_I_BDIFF,{"CFG_BIAS_SD_I_BDIFF",{{"CFG_BIAS_SD_I_BDIFF_Min",0},{"CFG_BIAS_SD_I_BDIFF_Max",255},}}},
+                {CFG_BIAS_SD_I_BFCAS,{"CFG_BIAS_SD_I_BFCAS",{{"CFG_BIAS_SD_I_BFCAS_Min",0},{"CFG_BIAS_SD_I_BFCAS_Max", 255},}}},
+                {CFG_BIAS_SD_I_BSF,{"CFG_BIAS_SD_I_BSF",{{"CFG_BIAS_SD_I_BSF_Min",0}, {"CFG_BIAS_SD_I_BSF_Max", 63},}}},
+                {CFG_BIAS_CFD_DAC_1,{"CFG_BIAS_CFD_DAC",{{"CFG_BIAS_CFD_DAC_1_Min",0},{"CFG_BIAS_CFD_DAC_1_Max", 63},}}},
+                {CFG_BIAS_CFD_DAC_2,{"CFG_BIAS_CFD_DAC",{{"CFG_BIAS_CFD_DAC_2_Min", 0},{"CFG_BIAS_CFD_DAC_2_Max", 63},}}},
+                {CFG_HYST,{"CFG_HYST",{{"CFG_HYST_Min",0},{"CFG_HYST_Max", 63},}}},
+                {CFG_THR_ARM_DAC,{"CFG_THR_ARM_DAC",{{"CFG_THR_ARM_DAC_Min",0},{"CFG_THR_ARM_DAC_Max", 255},}}},
+                {CFG_THR_ZCC_DAC,{"CFG_THR_ZCC_DAC",{{"CFG_THR_ZCC_DAC_Min",0}, {"CFG_THR_ZCC_DAC_Max", 255},}}},
+                {CFG_BIAS_PRE_VREF,{"CFG_BIAS_PRE_VREF",{{"CFG_BIAS_PRE_VREF_Min",0},{"CFG_BIAS_PRE_VREF_Max", 255},}}},
+                {CFG_VREF_ADC,{"CFG_VREF_ADC", {{"CFG_VREF_ADC_Min",0},{"CFG_VREF_ADC_Max", 3},}}}
+            };
+
+  
+
+
+        protected:
+
+        private:
+            /**
+             * @param classname is the class to check to see whether it is a GEMApplication inherited application
+             * @throws
+             */
+            bool isGEMApplication(const std::string& classname) const;
+
+            xdata::Integer m_nShelves;
+
+            log4cplus::Logger m_logger;
```

Comment:
In a previous iteration, this line had a comment to be removed, is that not the case?